.circleci/config.yml

```diff
@@ -8,146 +8,166 @@ jobs:
       - checkout
       - restore_cache:
           keys:
-            - cargo-v2-{{ checksum "Cargo.toml" }}-
-            - cargo-v2-
-      - run: cargo update
+            - cargo-v3-{{ checksum "Cargo.toml" }}
+            - cargo-v3-
+      - run: git submodule sync
+      - run: git submodule update --init
       - run: cargo fetch
       - persist_to_workspace:
-          root: "."
+          root: /mnt/crate
           paths:
             - Cargo.lock
       - save_cache:
-          key: cargo-v2-{{ checksum "Cargo.toml" }}-{{ checksum "Cargo.lock" }}
+          key: cargo-v3-{{ checksum "Cargo.toml" }}
```

Comment:
The point of doing it the old way was that the cache does not need to be re-saved if `Cargo.lock` did not change. `restore_cache` makes sure partial matches are considered.